components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -518,19 +515,20 @@ func (g *gpsrtk) getNtripFromVRS() error {
 
 	g.logger.Debugf("Writing GGA message: %v\n", ggaMessage)
 
-	_, err = g.vrsReaderWriter.WriteString(ggaMessage)
+	_, err = g.vrs.ReaderWriter.WriteString(ggaMessage)
 	if err != nil {
 		g.logger.Error("Failed to send NMEA data:", err)
 		return err
 	}
 
-	err = g.vrsReaderWriter.Flush()
+	err = g.vrs.ReaderWriter.Flush()
```

Comment:
Again, put these two paragraphs inside the `VRS` struct.

_edit: maybe even put these two paragraphs **and** starting the thread in a single function inside VRS!_

```go
func (vrs *VRS) WriteMessage(ggaMessage string, ggaFunc func() (string, error)) error {
    vrs.ReaderWriter.WriteString(ggaMessage)
    vrs.ReaderWriter.Flush()

    // ggaThread is StartGGAThread with, like, 2 levels of indentation removed.
    vrs.workers = utils.NewStoppableWorkers(vrs.ggaThread)
}
```

Reply:
make a function for WriteMessage/line, but I am not going to put the ggaThread function separate, mainly because I do not know if a user would always want to start the thread, or if a user wants to write a string that is not the gga message